File: frontend/src/app/core/models/ontology-model.ts
export interface OntologyModelRead {
  id: string;
  name: string;
  description: string;
  schema_id: string;
  schema_name: string;
  schema_version: number;
  class_count: number;
  instance_count: number;
  created_at?: string;
  updated_at?: string;
}

export interface OntologyModelCreate {
  name: string;
  description?: string;
  schema_id: string;
  schema_version?: number | null;
}

export interface OntologyModelUpdate {
  name?: string;
  description?: string;
}

File: backend/app/schemas/ontology_model.py
"""DTOs for named ontology models (schema version + instances)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class OntologyModelCreate(BaseModel):
    name: str = Field(min_length=1, max_length=128)
    description: str = ""
    schema_id: str
    schema_version: int | None = None


class OntologyModelUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=128)
    description: str | None = None


class OntologyModelRead(BaseModel):
    id: str
    name: str
    description: str = ""
    schema_id: str
    schema_name: str
    schema_version: int
    class_count: int = 0
    instance_count: int = 0
    created_at: datetime | None = None
    updated_at: datetime | None = None
